lib/BaseClient.ts

```diff
@@ -0,0 +1,22 @@
+import Logger from './Logger';
+import enums from './constants/enums';
+
+class BaseClient {
```

Comment:
should be an abstract class, assuming it's not meant to get instantiated directly